lib/Model/Pass/FixModel.cpp

```diff
@@ -31,6 +31,21 @@ void model::fixModel(TupleTree<model::Binary> &Model) {
       }
     }
 
+    // Filter out empty arrays.
+    bool IsEmptyArray = false;
+    for (const model::QualifiedType &QT : T->edges()) {
+      auto IsArray = [](const model::Qualifier &Q) {
+        return Q.Kind() == model::QualifierKind::Array;
+      };
+      auto Iterator = llvm::find_if(QT.Qualifiers(), IsArray);
+      if (Iterator != QT.Qualifiers().end() and !Iterator->Size()) {
+        ToDrop.insert(T.get());
```

Comment:
I feel like this function could be restructured as

```c++
  for (UpcastablePointer<model::Type> &T : Model->Types())
    if (shouldDrop(T))
      ToDrop.insert(T.get());
```

Outlining most of the logic will make the function simpler and reduce the amount of `continue`.